### PR TITLE
Fix determining viewLayer when using transitive dependency

### LINKED ISSUE #342: Could not find a supported Storybook viewlayer as transitive dependency
When using e.g. `@nuxtjs/storybook`, which has a [dependency](https://github.com/nuxt-community/storybook/blob/8306f371d4992c86f64da726c4c45a57daa64363/package.json#L29) on `@storybook/vue`, we fail to detect the viewlayer package.

Previously we would look in `node_modules` to detect the viewlayer, which would work for transitive dependencies. This was changed in #319 to look at `dependencies` in package.json directly.

### PR DESCRIPTION
**Pull request by ghengeveld**
Fixes #342 

This also drops `addons` reporting because we aren't using it and it's not reliable anyway.

### CODE REVIEW

File: bin/lib/getStorybookInfo.js [hunk 1]
```diff
@@ -23,59 +20,36 @@ const viewLayers = {
...
-  '@storybook/addon-queryparams': 'queryparams',
-  '@storybook/addon-storyshots': 'storyshots',
-  '@storybook/addon-storysource': 'storysource',
-  '@storybook/addon-viewport': 'viewport',
-};
+// Double inversion on Promise.all means fulfilling with the first fulfilled promise, or rejecting
+// when _everything_ rejects. This is different from Promise.race, which immediately rejects on the
+// first rejection.
+const invert = (promise) => new Promise((resolve, reject) => promise.then(reject, resolve));
+const raceFulfilled = (promises) => invert(Promise.all(promises.map(invert)).then((arr) => arr[0]));
```
Comment (tmeasday) [inline bin/lib/getStorybookInfo.js:29]: Lol neat trick

**Review by tmeasday — CHANGES_REQUESTED** [review]
This LGTM apart from dropping addons -- we do use the fact that you are using `docs` or `essentials` to decide whether to show the docs tab on the component screen.

We'll need to change that in Chromatic before we can drop it.

File: bin/lib/getStorybookInfo.js (hunk 1 above)
Comment (ghengeveld) [reply bin/lib/getStorybookInfo.js:29]: It was nice to flex my async muscles again.

**Comment by ghengeveld** [comment]
> This LGTM apart from dropping addons -- we do use the fact that you are using `docs` or `essentials` to decide whether to show the docs tab on the component screen.
> 
> We'll need to change that in Chromatic before we can drop it.

Right, I wasn't aware of that. Reverted.

**Pull request merged**